cmake/checks/check_01_compiler_features.cmake

```diff
@@ -317,6 +319,24 @@ ENDIF()
 
 
 #
+# Do a similar check with the always_inline attribute on functions.
+#
+CHECK_CXX_SOURCE_COMPILES(
+  "
+          __attribute__((always_inline)) int fn () { return 0; }
+          int main () { return fn(); }
+  "
+  DEAL_II_COMPILER_HAS_ATTRIBUTE_ALWAYS_INLINE
+  )
+
+IF(DEAL_II_COMPILER_HAS_ATTRIBUTE_ALWAYS_INLINE)
+  SET(DEAL_II_ALWAYS_INLINE "__attribute__((always_inline))")
```

Comment:
With gcc-5.3.0 I run into quite a number of warnings issued by ```-Wattributes```
```
warning: always_inline function might not be inlinable [-Wattributes]
```
I suggest that we add
```
ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-attributes")
```
to suppress these warnings.